### DISCUSSION TITLE
Where can be a good place to start?

### DISCUSSION THREAD

**Discussion by logic-finder** [WebAudio/web-audio-api#2460] [General]
Hello! nice to meet you. I think many audio experts are here, so if you don't mind me asking, I would like to ask a little bit long question.

I'm a person who do love computer programming but sadly I do not know much about actual programming and language usage. Currently I'm learning JavaScript and highly interested in Web Audio API.

Lately, I met a problem which I failed and failed in college admissions and I think perhaps it's because of my low GPA. However, I still want to learn about digital sound, digital signal processing, speech synthesis, artificial intelligence related to sound. Someone may say, "then challenge to next year's admission', but I think I'm quite older than current high school students, so it would be difficult(?) to prepare college entrance exam (like SAT). (What makes me worried about is to waste time just for preparing it)

My math knowledge is at basic Calculus (I studied AP Calculus AB in high school and got 5 on the exam). I am highly interested in generating sounds using computers, but know **nothing** about it. I want to learn about these things I wrote above, but **I have no idea about where to start**: a structure of MIDI files? characteristics of waves? Fourier transform? a history of digital sound?

Currently I have been translating Web Audio API documents into Korean in MDN Web Docs (for the sake of my studying) and I'm trying to read the whole W3C Web Audio API Specification. Actually, as English is not my first language and I'm not a fluent speaker, it is a hard thing for me to read English texts; nevertheless I'm as firmly determined as I can overcome those adversities; since I want to know about sounds.

I still wonder how it would feel like being a college student and what place would be a graduate school, but I think such things do not exist in my destiny. I have a curiosity how students in computer science division are taught on digital sound in common?

It would be appreciated if you give me an advice.
Thank you very, very much for reading this long question.
_logic-finder_

**Comment by bradisbell** [comment]
Greetings, @logic-finder.

Firstly, it's never too late to further your education if you choose to.  I don't know where you are located, but many schools will accommodate non-English speakers, and will also not require high GPAs.  It might be worth studying up on whichever topics gave you troubles, or considering other schools.

In any case, neither programming nor audio require college education!  And, based on the topics you're asking about, it seems that you actually have some knowledge of the domain.  The Web Audio API is a great place to start.  It has everything you need to make anything from simple synthesizers, all the way up to full DAW applications.

Perhaps something to start with is the basics of synthesizers in general.  Ableton has an excellent tutorial on this:  https://learningsynths.ableton.com/  Note that this entire tutorial website is made using the Web Audio API, so everything you can do there, you can write code for eventually.  If you want to get started making your own synthesizer, start with the OscillatorNode:  https://developer.mozilla.org/en-US/docs/Web/API/OscillatorNode

You've probably already seen it, but the MDN docs have a good tutorial for playing back some basic audio as well:  https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API/Using_Web_Audio_API

The W3C specification for the Web Audio API is probably not the best place to start.  I don't want to discourage you from reading it, but it isn't meant to be a reference to end users as much it is an agreement between browser developers on what the standards should be.

This GitHub board is meant to be a discussion on those specifications, and isn't really meant to be a place to find other like-minded folks, nor answer questions about the usage of the Web Audio API.  I'd encourage you to check out Stack Overflow as you have questions about using the Web Audio API.

I hope you find this information helpful, and continue learning about the audio APIs available.  Browsers are great for this, and you're starting out at a particularly good time in the web, where browsers are very capable application platforms.

Good luck!

-Brad Isbell (AudioPump, Inc.)

**Reply by logic-finder** [reply]
Thank you for letting me know these good informations! Your advice really helped and encouraged me. Especially Ableton seems really interesting. Thank you again.

p.s. I would like to apologize for choosing wrong place to ask a question.

**Reply by logic-finder** [reply]
Wow! Ableton tutorials was extraordinarily fantastic!!! I just can't express my gratitude with usual 'thank you' expression!
